From: Zhenhua Huang <quic_zhenhuah@quicinc.com>
Subject: mm/page_owner.c: remove redudant drain_all_pages
Date: Wed, 7 Sep 2022 16:01:13 +0800

Page owner info of pages in pcp list has already been reset:

	free_unref_page
		-> free_unref_page_prepare
			-> free_pcp_prepare
				-> free_pages_prepare which does page owner
				   reset
		-> free_unref_page_commit which add pages into pcp list

It can also be confirmed from dump that page owner info of pcp pages are
correct.  Hence there is no more need to drain when reading.

Link: https://lkml.kernel.org/r/1662537673-9392-1-git-send-email-quic_zhenhuah@quicinc.com
Signed-off-by: Zhenhua Huang <quic_zhenhuah@quicinc.com>
Cc: Mel Gorman <mgorman@techsingularity.net>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/page_owner.c |    2 --
 1 file changed, 2 deletions(-)

--- a/mm/page_owner.c~mm-page_ownerc-remove-redudant-drain_all_pages
+++ a/mm/page_owner.c
@@ -524,8 +524,6 @@ read_page_owner(struct file *file, char
 	while (!pfn_valid(pfn) && (pfn & (MAX_ORDER_NR_PAGES - 1)) != 0)
 		pfn++;
 
-	drain_all_pages(NULL);
-
 	/* Find an allocated page */
 	for (; pfn < max_pfn; pfn++) {
 		/*
_
